### DISCUSSION TITLE
Winget does not display exe instead of displaying MSI while the Scope is in Machine

### DISCUSSION THREAD

**Discussion by BrandonWanHuanSheng** [microsoft/winget-pkgs#108175] [Q&A] [answered]
When I run winget show 7zip.7zip. 
![Screenshot (1158)](https://github.com/microsoft/winget-pkgs/assets/118059864/d14d690b-3978-4f70-94a7-a4dca2b51fa5)
![Screenshot (1159)](https://github.com/microsoft/winget-pkgs/assets/118059864/a9a81867-b42f-4818-b428-daa02f6cba97)
![Screenshot (1160)](https://github.com/microsoft/winget-pkgs/assets/118059864/7e506f17-8fee-47f3-a85c-2595a8b89db1)
It look like it is not displaying exe applications while Scope is set to Machine. It look like winget still hiding exe installer. But in the manifest, I saw some exe application with MSI together. Does need additional commands for winget?

**Comment by mdanish-kh** [comment]
`winget show` does not show the complete manifest but rather a subset of the manifest metadata that best matches your OS architecture, locale, installation scope and the installer type of the current installation. The manifest does contain both EXE and MSI installers but `winget show` shows you the installer that you're going to get if you run `winget upgrade` / `winget install`

**Reply by BrandonWanHuanSheng** [reply]
How do I install 7-zip.7zip exe application when the Scope set to Machine?

**Reply by mdanish-kh** [reply, marked as answer]
You can't specify the installer type for winget install. There is an open issue at winget-cli tracking the request: 
- https://github.com/microsoft/winget-cli/issues/1166

In the upgrade scenario though, WinGet will try to match the existing installation with a suitable InstallerType present in the manifest. If your existing installation was performed using an EXE installer, WinGet will use the EXE installer from the manifest to serve subsequent upgrades.
 So, if you really want WinGet to use the EXE installer for future upgrades, you may need to perform a manual installation first using the EXE
